data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -120,267 +72,100 @@ def get_prices_range(collateral_token_name: str, current_price: Decimal) -> Iter
         """
-        Gets the response from endpoint for the given protocol.
+        Calculates liquidable debt based on data provided and updates an existing data.
+        Data to calculate liquidable debt for:
         :param protocol_name: str
-        :return: requests.Response
+        :return: A dictionary of the ready liquidable debt data.
         """
-        response = requests.get(cls.INTEREST_MODEL_VALUES_URL.format(protocol=protocol_name))
-        response.raise_for_status()
-        return response.json()
+        data, interest_rate_models = self.fetch_data(protocol_name=protocol_name)
+        state = self.state_class()
 
-    @classmethod
-    def get_interest_rate_models_data(cls, protocol_name: str) -> dict:
-        """
-        Returns an interest rate models data
-        :param protocol_name: str
-        :return: dict
-        """
-        cls.verify_endpoint_protocol(protocol_name)
-        for protocol in AvailableProtocolID:
-            current_protocol = str(protocol.value).lower()
-            if protocol_name.lower() == current_protocol:
-                try:
-                    return cls._get_response(protocol.value)
-                except:
-                    time.sleep(10)
-                    return cls._get_response(protocol.value)
+        for instance in data:
+            loan_entity = ZkLendLoanEntity()
 
-    @staticmethod
-    def normalize_protocol_name(protocol_name: str) -> str:
-        """
-        Normalizes the given protocol name.
-        :param protocol_name: str
-        :return: str
-        """
-        return "_".join(protocol_name.split(" ")).lower()
+            loan_entity.debt = TokenValues(values=instance.debt)
+            loan_entity.collateral = TokenValues(values=instance.collateral)
 
-    def _calculate_liquidable_debt(
-            self, data: dict = None,
-    ) -> dict:
-        """
-        Calculates liquidable debt based on data provided and updates an existing data.
-        Data to calculate liquidable debt for:
-        :param data: {
-            Row number: {
-                Field name: Token value
-                 or Decimal - Field value (the price as example)
-                 or {
-                    Token name: Token value
+            state.loan_entities.update(
+                {
+                    instance.user: loan_entity,
                 }
-            }
-        }
-        :return: A dictionary of the ready liquidable debt data.
-        """
-        # FIXME Read loan_entities from the database
-        # FIXME concentrate for one Protocol and one collateral_token and debt_token
-        # FIXME go through all data and init them as loan_entities
-        result_data = dict()
+            )
+
+        # Set up collateral and debt interest rate models
+        state.collateral_interest_rate_models = InterestRateModels(
+            init_value=Decimal(interest_rate_models.collateral.get("STRK", ""))
+        )
```

Comment:
These should contain interest rate indices for all tokens (STRK, ETC, WBTC, USDC, USDT, ...), not just STRK.